core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -189,7 +191,17 @@ public ListFeatureSetsResponse listFeatureSets(ListFeatureSetsRequest.Filter fil
     ListFeatureSetsResponse.Builder response = ListFeatureSetsResponse.newBuilder();
     if (featureSets.size() > 0) {
       for (FeatureSet featureSet : featureSets) {
-        response.addFeatureSets(featureSet.toProto());
+        if (labels != null && !labels.isEmpty()) {
```

Comment:
This can be made more concise by using guard clauses. 
```
If( label filter ){
test if label in feature set, if not, "continue"
}
response.addFeatureSets(...);
```

Reply:
Updated.